src/modules/appeals-service/AppealsService.ts

```diff
@@ -37,6 +42,38 @@ export class AppealsService {
 
     }
 
+    public async hasExistingAppeal(companyNumber: string, penaltyReference: string, accessToken: string,
+                                   refreshToken: string): Promise<boolean> {
+
+        const operation = 'get';
+
+        this.checkArgumentOrThrow(companyNumber, 'Company number is missing');
+        this.checkArgumentOrThrow(penaltyReference, 'penalty reference is missing');
+        this.checkArgumentOrThrow(accessToken, 'Access token is missing');
+        this.checkArgumentOrThrow(refreshToken, 'Refresh token is missing');
+
+        this.refreshTokenInterceptor(accessToken, refreshToken);
+
+        const uri: string = `${this.uri}/companies/${companyNumber}/appeals?`;
+        loggerInstance()
+            .debug(`Making a GET request to ${uri}`);
+
+        const params = {penaltyReference};
+
+        try {
+            await this.axiosInstance.get(uri, {params});
```

Comment:
is it worth checking the status of the response and if there is data before we return 'true'?